### PR TITLE
London_10/Anu Thapaliya/Full-Stack-Project-Assessment/level200/250/299

### PR DESCRIPTION
**Pull request by anuthapaliy**

### CODE REVIEW

**Comment by kitko112** [comment]
add node_modules to .giitignore

File: client/src/AddNewVideo.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+import axios from 'axios';
+
+
+const AddNewVideo = ({ addVideoButton }) => {
+const [title, setTitle] = useState('');
+const [url, setUrl] = useState('');
+// const [error, setError] = useState('');
+
+
+// const correctYoutubeUrl = (url) => {
```
Comment (illicitonion) [inline client/src/AddNewVideo.js:13]: Please try to remove commented out code in the future :)

File: client/src/AddNewVideo.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+// if (!correctYoutubeUrl(url)) {
+// setError('Invalid youTube URL.');
+// return;
+// }
+
+
+// setError('');
+const handleAddVideo = async () => {
+try {
```
Comment (illicitonion) [inline client/src/AddNewVideo.js:46]: I'd generally recommend formatting your code with indentation, it'll make it much easier to read and follow, and to debug things like extra/missing brackets.

File: client/src/AddNewVideo.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+// if successful, fetch updated videos and clear input fields
+// fetchVideos();
+// setTitle('');
+// setUrl('');
+// }
+// } catch (error) {
+// console.error('Error adding video:', error);
+// }
+// };
+addVideoButton(newVideo);
```
Comment (illicitonion) [inline client/src/AddNewVideo.js:72]: It's quite strange to call a function "button" - can you think of a name for this variable which better expresses what it does and how it's used?

File: client/src/App.js [hunk 4]
```diff
@@ -1,13 +1,193 @@
...
+console.log('Error fetching videos:', error);
+}
+}, [order]);
+
+
+
+
+useEffect(() => {
+fetchVideos();
+}, [fetchVideos]);
```
Comment (illicitonion) [inline client/src/App.js:43]: What does adding `fetchVideos` to this array do? What would be different if you removed it?

File: client/src/App.js [hunk 5]
```diff
@@ -1,13 +1,193 @@
...
+
+} catch (error) {
+console.error('Error deleting video:', error)
+}
+};
+
+
+const handleAddVideo = async (newVideo) => {
+if (newVideo.title.trim() === '' || !isValidYouTubeUrl(newVideo.url)) {
+return;
```
Comment (illicitonion) [inline client/src/App.js:119]: If this happens, what will the user see? How will they know what they should do next?

File: client/src/VideoDisplay.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+
+const VideoDisplay = ({ video, handleUpVote, handleDownVote, handleRemoveVideo }) => {
+return (
+<div className='video-card'>
+<h2>{video.title}</h2>
+<iframe
+width="560"
+height="315"
+src={`https://www.youtube.com/embed/${video.url.split('v=')[1]}`}
```
Comment (illicitonion) [inline client/src/VideoDisplay.js:14]: What would happen if the video URL didn't contain `v=`? Is there a way we could handle that better?

File: server/server.js [hunk 7]
```diff
@@ -1,15 +1,352 @@
+<<<<<<< HEAD
```
Comment (illicitonion) [inline server/server.js:1]: It looks like you had a merge conflict here that didn't get resolved?

**Review by illicitonion — COMMENTED** [review]
It'd be really handy to make sure your PRs are easy to review in the future - please think about node_modules, indentation, etc. Next time you make a PR, have a look at the "Files Changed" tab and make sure it's easy to read! :)

**Pull request closed without merging**